app/Http/Controllers/AssignmentController.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+class AssignmentController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        return inertia("AdminPage");
+    }
+
+
+    public function create(User $user): RedirectResponse {
+        if ($user->is_admin) {
+            return redirect()->route('assignment.index')->with('success', 'Assignment created');
+        }
+        return redirect()->route('assignment.index')->with('error', 'User is not an admin');
+    }
```

Comment:
This doesn't make an assignment which it should. Also any user should be able to make an assignment for their class rather than just admins. Your policies should control whether a user can access or not.